client/lib/purchases/index.ts

```diff
@@ -939,6 +939,7 @@ export function purchaseType( purchase: Purchase ) {
 }
 
 export function getRenewalPrice( purchase: Purchase ) {
+	// TODO clk this returns non-integer values, which breaks comparisons/operations with plan integer-based pricing
 	return purchase.saleAmount || purchase.amount;
```

Comment:
@sirbrillig / @Automattic/shilling I recall we came across similar scenario previously. The non-integer values returned from `getRenewalPrice` are not comparable with the integer-based plan pricing computed for the `relatedMonthlyPlanPrice`.

I can see there are `saleAmountInteger` and `priceInteger` properties on the `purchase` object, but no `amountInteger`. Is it safe to use the former in place of `saleAmount` and `amount`?

Reply:
@chriskmnds  @sirbrillig is on sabbatical, but I am going to take a look for you. If I can't confirm, then I'll check with the rest of the team.

Reply:
Looking at `getRenewalPriceInSmallestUnit` on lines 945-947, it seems that this function already uses `saleAmountInteger` and `priceInteger`. Can you test and see if this works for your case? I think the problem you might run into is that these properties return the purchase amount in cents ( 4800 instead of 48). So your current math would need adjusting because 4800/12 != 48/12

However, when I look at the purchase property, it seems that `amount` should be an integer? So I am not clear in understanding why `getRenewalPrice()` would not return an integer.

Something else to consider, I don't think we would ever want to show a sale price on the subscription management page. It should always be the full price of the product, because, without other intervention, that is what the customer would renew for. If that is the case, you can skip calling `getRenewalPrice()`.

Maybe replace line 458 in `renderRenewAnnualNavItemI()` with this instead:

`const annualPrice = purchase.amount / 12;`

Let me know if this helps or if you have any other questions!

Reply:
> Is it safe to use the former in place of saleAmount and amount?

I guess that depends if we ever display floats in our pricing? If we ever have a situation where the renewal price is actually a float, and we display it as an integer yet charge it on the backhand as a float, then we may run into an issue where we aren't displaying the correct data. 🤔 

@DavidRothstein may know more about this, but my gut is telling me that we need to be cautious here as we do not know what the future of pricing may look like and what units we'll need to display.

And as I look closer, it looks like we use this function to set the `rawPrice` in some instances, which would imply there is no rounding involved. I wonder if it makes more sense to create a separate function like:

```
export function getRenewalPriceInteger( purchase: Purchase ) {
	return purchase.saleAmountInteger || purchase.priceInteger;
}
```

> However, when I look at the purchase property, it seems that amount should be an integer? So I am not clear in understanding why getRenewalPrice() would not return an integer.

It looks like the `amount` is a float which is set here - fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sjcpbz%2Qovyyvat%2Sfgber%2Qfhofpevcgvba.cuc%3Se%3Q1qoq911r%233773-og - in the API. 

That figure is generated here - fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sjcpbz%2Qovyyvat%2Spynff.jcpbz%2Qovyyvatqnqql.cuc%3Se%3Q0qo9sqon%23937-og - which returns a float.

`priceInteger` does a bit of rounding to ensure an integer is returned, though it uses the same base subscription code linked above.

Reply:
Thank you both for the wonderful insight. 🙇  It looks like switching to `getRenewalPriceInSmallestUnit` (which i missed seeing!) fixes the issue. It works because `relatedMonthlyPlanPrice` used in manage-purchase component is in the smallest unit too. So I think we are good. 

I'm not sure about whether it is correct to use `saleAmount` or not. It makes sense to not involve a sale in this comparison, although this is how it's done in production. Perhaps something for a separate issue to address. Would you like me to create an issue and focus on this separately? 

For the moment, the % savings shown look same as production. Img below with an expiring Personal plan - `55%` savings is correct now):

<img width="600" alt="Screenshot 2024-07-23 at 3 11 59 PM" src="https://github.com/user-attachments/assets/40da5e17-cb97-40b3-86c1-1ba5ced181df">

Reply:
> I'm not sure about whether it is correct to use saleAmount or not. It makes sense to not involve a sale in this comparison, although this is how it's done in production. Perhaps something for a separate issue to address. Would you like me to create an issue and focus on this separately?

I mean this concerning @JessBoctor's comment from above:

> Something else to consider, I don't think we would ever want to show a sale price on the subscription management page. It should always be the full price of the product, because, without other intervention, that is what the customer would renew for. If that is the case, you can skip calling getRenewalPrice().

Ultimately, we could indeed just compare the plan's monthly and yearly prices and not involve the purchased amount at all. Is this what you mean? I'd happily address this concern here or in an immediatre follow-up.

Reply:
> Ultimately, we could indeed just compare the plan's monthly and yearly prices and not involve the purchased amount at all. Is this what you mean? I'd happily address this concern here or in an immediatre follow-up.

@chriskmnds Let me get some feedback from the team.

The way I think about it, the manage purchase screen is all about prices going forward. We wouldn't want to include something like a sale price here because a user may not actually qualify for it, or it might change, I think it would be better to be more conservative with our offering (55% vs 75%) and give the customer an extra surprise if, for some reason, a sale is applied. This is a better experience than the customer expecting a higher discount with the upgrade and not receiving it at Checkout.

Reply:
Thanks. Let me know and I can apply change.

> I think it would be better to be more conservative with our offering (55% vs 75%) and give the customer an extra surprise if, for some reason, a sale is applied.

Indeed, this feels better to me too. So we can resolve to not using the sale amount for the base price in that case.

Reply:
@chriskmnds Thanks for your patience! @Automattic/shilling talked about this in our team call yesterday. The consensus is that it _would be better_ to use the `getRenewalPriceInSmallestUnit` function rather than the `purchase.amount`. The expectation is that `getRenewalPriceInSmallestUnit` should only return any sale prices which the user would be eligible for. If the function isn't doing that, then we should address that as a separate bug and fix it, rather than proactively avoiding it here.

Reply:
Thanks @JessBoctor ! I appreciate you looking into this with the team. We can continue with `getRenewalPriceInSmallestUnit` then 👍